src/cure.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+/// cure.sol -- Debt Rectifier contract
+
+// Copyright (C) 2022 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.6.12;
+
+interface SourceLike {
+    function cure() external view returns (uint256);
+}
+
+contract Cure {
+    mapping (address => uint256) public wards;
+    uint256 public live;
+    address[] public sources;
+    uint256 public total;
+    mapping (address => Source) public data;
+
+    struct Source {
+        uint128 pos;
+        uint128 amt;
```

Comment:
mm this is a RAD (at least treated as such in END), so cannot be store as uint128.

Reply:
To avoid the use of uint128 we can use a separate positions array.
Will be more gas costly but I don't think it matters so much specifically in this contract.

Reply:
Done.